numpy/__init__.pyi

```diff
@@ -778,6 +779,21 @@ class nditer:
         buffersize: Any = ...,
     ) -> Any: ...
     def __getattr__(self, key: str) -> Any: ...
+    def __enter__(self) -> nditer: ...
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc_value: Optional[BaseException],
+        traceback: Optional[TracebackType]
+    ) -> Optional[bool]: ...
+    def __iter__(self) -> Iterator[Any]: ...
+    def __next__(self) -> Any: ...
+    def __len__(self) -> int: ...
+    def __copy__(self) -> nditer: ...
+    def __getitem__(self, index: int) -> Any: ...
+    def __setitem__(self, index: int, value: Any) -> None: ...
+    def __delitem__(self, key: int) -> None: ...
```

Comment:
```suggestion
    def __getitem__(self, index: SupportsIndex | slice) -> Any: ...
    def __setitem__(self, index: SupportsIndex | slice, value: Any) -> None: ...
    def __delitem__(self, key: SupportsIndex | slice) -> None: ...
```
The `__<x>item__` dunders can also take `slice` objects or any object implementing the `__index__` methods (not just `int`).